tests/mock/vscode.ts

```diff
@@ -918,6 +956,25 @@ export class VSCode {
       this._webviewProviders.set(name, provider);
       return disposable;
     };
+    this.window.createWebviewPanel = (viewType: string) => {
+      const { webview, pagePromise } = this._createWebviewAndPage();
+      const didDispose = new EventEmitter<void>();
+      const panel: any = {};
+      panel.onDidDispose = didDispose.event;
+      panel.webview = webview;
+      pagePromise.then(webview => {
+        webview.on('close', () => panel.dispose());
```

Comment:
I don't see webview.on('close') in VS Code so this is your invention it seems. What are you trying to do here? I understand how panel.dispose() should close webview and page, but what is this one about?

Reply:
It's to simulate closing the webview panel manually. I use it here

https://github.com/microsoft/playwright-vscode/blob/2c1d2e7972c1944926ff92a7daf21a5f955bf81e/tests/trace-viewer.spec.ts#L119